### PR TITLE
Add and update opensearch-devops related documents

### LINKED ISSUE #6: Level up README and related markdown content
Level up and cleanup markdown content. 

See [.github#8](https://github.com/opensearch-project/.github/issues/8) for details and recommendations, and [common-utils](https://github.com/opensearch-project/common-utils/) for an example.

- [x] README.md
- [x] CONTRIBUTING.md
- [x] DEVELOPER_GUIDE.md
- [x] MAINTAINERS.md
- [x] ADMINS.md
- [x] SECURITY.md
- [x] RELEASING.md
- [x] TESTING.md
- [x] other .md files

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add and update opensearch-devops related documents
 
### Issues Resolved
#6
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+OpenSearch (https://opensearch.org/)
```
Comment (TheAlgo) [inline NOTICE.txt:1]: After this will he have license headers for every file?

Comment (peterzhuamazon) [reply NOTICE.txt:1]: https://github.com/opensearch-project/.github/pull/29

Comment (peterzhuamazon) [reply NOTICE.txt:1]: Yes, all the new files needs to add these two lines.
```
New files in your code contributions should contain the following license header. If you are modifying existing files with license headers, or including new files that already have license headers, do not remove or modify them without guidance.
```

```
# Copyright OpenSearch Contributors
# SPDX-License-Identifier: Apache-2.0
```

Comment (TheAlgo) [reply NOTICE.txt:1]: Ok so we will only add it in the new files that we create and keep the old ones as it is?

Comment (peterzhuamazon) [reply NOTICE.txt:1]: I think it means all the old files WITH EXISTING HEADER can stay the same, means files without headers needs to get added with the new ones.

File: MAINTAINERS.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## Maintainers
+
+| Maintainer | GitHub ID | Affiliation |
+| --------------- | --------- | ----------- |
+| Barani Bikshandi | [bbarani](https://github.com/bbarani) | Amazon |
+| Peter Zhu | [peterzhuamazon](https://github.com/peterzhuamazon) | Amazon |
+| Peter Nied | [peternied](https://github.com/peternied) | Amazon |
+| Sayali Gaikawad | [gaiksaya](https://github.com/gaiksaya) | Amazon |
```
Comment (DandyDeveloper) [inline MAINTAINERS.md:8]: Am I able to get added to this list? I'll get more notifications and such this way.

**Review by DandyDeveloper — APPROVED** [review]
I'll likely be adding heaps and have will continue to answer issues and make PRs. 

I'd appreciate being added as a maintainer. If not, that's fine.

Everything else looks good.

**Review by TheAlgo — APPROVED** [review]

File: MAINTAINERS.md (hunk 2 above)
Comment (peterzhuamazon) [reply MAINTAINERS.md:8]: Hi @DandyDeveloper,

We are currently getting a decision out for adding external maintainers. Please stay tune for more updates. If we eventually add you there will be another PR to add your name here.

In the meantime please feel free to read the maintainer file for your responsibility:
https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md

Thanks.

**Review by zelinh — APPROVED** [review]

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -1,4 +1,22 @@
...
+* Offensive comments related to gender, gender identity and expression, sexual orientation, disability, mental illness, race, political or religious affiliation;
+* Posting of sexually explicit or violent content;
+* The use of sexualized language and unwelcome sexual attention or advances;
+* Public or private harassment of any kind;
+* Publishing private information, such as physical or electronic address, without permission;
+* Other conduct which could reasonably be considered inappropriate in a professional setting;
+* Advocating for or encouraging any of the above behaviors.
+* Enforcement and Reporting Code of Conduct Issues:
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported. [Contact us](mailto:opensource-codeofconduct@amazon.com). All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances.
```
Comment (camerski) [inline CODE_OF_CONDUCT.md:22]: Should we just link to https://github.com/opensearch-project/.github/blob/main/CODE_OF_CONDUCT.md ?

**Review by camerski — APPROVED** [review]

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (peterzhuamazon) [reply CODE_OF_CONDUCT.md:22]: This is copied from @dblock contribution to opensearch-build repo.
If needed I can change to a link.

Comment (peterzhuamazon) [reply CODE_OF_CONDUCT.md:22]: In another PR.

**Pull request merged**